src/NetTopologySuite/Operation/OverlayNG/RingClipper.cs

```diff
@@ -0,0 +1,192 @@
+﻿using NetTopologySuite.Geometries;
+
+namespace NetTopologySuite.Operation.OverlayNg
+{
+    /// <summary>
+    /// Clips a ring of points to an rectangle.
+    /// Uses a variant of Cohen-Sutherland clipping.
+    /// <para/>
+    /// In general the output is not topologically valid.
+    /// In particular, the output may contain coincident non-noded line segments
+    /// along the clip rectangle sides.
+    /// However, the output is sufficiently well-structured
+    /// that it can be used as input to the <see cref="OverlayNG"/> algorithm
+    /// (which is able to process coincident linework due
+    /// to the need to handle topology collapse under precision reduction).
+    /// <para/>
+    /// Because of the likelihood of creating
+    /// extraneous line segments along the clipping rectangle sides,
+    /// this class is not suitable for clipping linestrings.
+    /// <para/>
+    /// The clipping envelope should be generated using <see cref="RobustClipEnvelopeComputer"/>,
+    /// to ensure that intersecting line segments are not perturbed
+    /// by clipping.
+    /// This is required to ensure that the overlay of the
+    /// clipped geometry is robust and correct (i.e. the same as
+    /// if clipping was not used).
+    /// </summary>
+    /// <seealso cref="LineLimiter"/>
+    /// <author>Martin Davis</author>
+    public class RingClipper
```

Comment:
this isn't designed to be inherited from, so let's mark it `sealed`

(We can always unseal it later if that becomes a problem)